index.ts

```diff
@@ -0,0 +1,60 @@
+const products = require("./data/products.json");
+
+interface Product {
+  name: string;
+  description: string;
+  price: number;
+  image: URL;
+  limit: number;
```

Comment:
does all products have a limit?

Reply:
no. i changed it to a partial typeartial type